osu.Game/Rulesets/Difficulty/Utils/DecayingValue.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Diagnostics;
+
+namespace osu.Game.Rulesets.Difficulty.Utils
+{
+    /// <summary>
+    /// Represents a value that exponentially decays over time
+    /// </summary>
+    public class DecayingValue
```

Comment:
Understanding this class (and it's uses everywhere) clicked for me when I realised this was representing strain. I think renaming it to retain the terminology (something similar to `DecayingStrain` or even `Strain`) would go a long way - if not then certainly include it in the xmldoc.

Reply:
Case in point, it would be a lot more intuitive for me to have methods such as `IncrementStrainAtTime` and `StrainValueAtTime` - but this could easily be because I'm used to the current lingo.

Reply:
I think this is just familiarity. You can type "decaying value" or "exponential decay" into google and the first link is a wikipedia article decribing the exact concept.